source/chip/STM32/STM32L0/STM32L0-PWR.cpp

```diff
@@ -30,7 +30,12 @@ int configureVoltageScaling(const uint8_t voltageScale)
 	if (voltageScale < minVoltageScale || voltageScale > maxVoltageScale)
 		return EINVAL;
 
-	PWR->CR = (PWR->CR & ~PWR_CR_VOS) | (((4 - voltageScale) << PWR_CR_VOS_Pos) & PWR_CR_VOS);
+	while ((PWR->CSR & PWR_CSR_VOSF) == 1);	// wait until VOSF bit is reset to 0
```

Comment:
Assuming that the user will use provided functions for adjusting voltage scaling, I would say the first wait is not necessary and can be skipped. I wrote a similar code for STM32L4 (not published yet) and decided to stay with just the wait after the change.

Reply:
Perhaps you have right. I've just want to be fully in accordance with STM32L0 reference manual where steps 2 and 4 requiring polling VOSF bit until is reset to 0.
I didn't noticed similar requirement to wait before voltage scaling in STM32L4 reference manual, so are you sure that change will be correct?